chainstate/test-suite/src/tests/pos_accounting_reorg.rs

```diff
@@ -0,0 +1,171 @@
+            .add_input(
+                TxInput::new(
+                    OutPointSourceId::Transaction(tx_b.transaction().get_id()),
+                    0,
+                ),
+                empty_witness(&mut rng),
+            )
+            .add_output(TxOutput::new(
+                OutputValue::Coin(Amount::from_atoms(rng.gen_range(1000..100_000))),
+                OutputPurpose::StakePool(Box::new(StakePoolData::new(
+                    anyonecanspend_address(),
+                    None,
+                    vrf_pub_key_c,
+                    pub_key_c,
+                    0,
+                    Amount::ZERO,
+                ))),
+            ))
+            .build();
+
+        // create block a
+        let block_a = tf
+            .make_block_builder()
+            .add_transaction(tx_a)
+            .build_and_process()
+            .unwrap()
+            .unwrap();
+        assert_eq!(
+            tf.best_block_id(),
+            Id::<GenBlock>::from(*block_a.block_id())
+        );
+
+        // create block b
+        let block_b = tf
+            .make_block_builder()
+            .with_parent(genesis_id.into())
+            .add_transaction(tx_b.clone())
+            .build();
+        let block_b_id = block_b.get_id();
+        tf.process_block(block_b, BlockSource::Local).unwrap();
+
+        // no reorg here
+        assert_eq!(
+            tf.best_block_id(),
+            Id::<GenBlock>::from(*block_a.block_id())
+        );
+
+        // create block c
+        let block_c = tf
+            .make_block_builder()
+            .with_parent(block_b_id.into())
+            .add_transaction(tx_c.clone())
+            .build_and_process()
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(
+            tf.best_block_id(),
+            Id::<GenBlock>::from(*block_c.block_id())
+        );
+
+        // Accounting data in storage after reorg should equal to the data in storage for chain
+        // where reorg never happened.
+        //
+        // Construct fresh `genesis -> b -> c` chain as a reference
+        let expected_accounting_data = {
+            let storage = Store::new_empty().unwrap();
+            let mut tf = TestFramework::builder(&mut rng)
+                .with_storage(storage.clone())
+                .with_chainstate_config(tf.chainstate().get_chainstate_config())
+                .build();
+
+            let block_b = tf
+                .make_block_builder()
+                .with_parent(genesis_id.into())
+                .add_transaction(tx_b)
+                .build_and_process()
+                .unwrap()
+                .unwrap();
+
+            tf.make_block_builder()
+                .with_parent((*block_b.block_id()).into())
+                .add_transaction(tx_c)
+                .build_and_process()
+                .unwrap();
+
+            storage.read_accounting_data()
+        };
+
+        assert_eq!(storage.read_accounting_data(), expected_accounting_data);
```

Comment:
I haven't replaced this with full storage comparison, because reorg affects db leaving traces in DBBlock and DBBlockIndex. So I cannot compare chain a->b->c that was reorged with iconic chain a->b->c that was not.

I took this approach because I couldn't find other reasonable way to construct full storage without processing blocks.

Reply:
Can we exclude these, or even better, add them manually to the result?

Reply:
done